pkg/adaptor/proxy/factory.go

```diff
@@ -3,27 +3,52 @@
 
 package proxy
 
-import "github.com/confidential-containers/cloud-api-adaptor/pkg/util/tls"
+import "github.com/confidential-containers/cloud-api-adaptor/pkg/util/tlsutil"
 
 type Factory interface {
-	New(sandboxID string) AgentProxy
+	New(serverName, socketPath string) AgentProxy
 }
 
 type factory struct {
-	PauseImage    string
-	CriSocketPath string
-	TLSConfig     *tls.TLSConfig
+	pauseImage    string
+	criSocketPath string
+	tlsConfig     *tlsutil.TLSConfig
+	caService     tlsutil.CAService
 }
 
-func NewFactory(pauseImage, criSocketPath string, tlsConfig *tls.TLSConfig) Factory {
+func NewFactory(pauseImage, criSocketPath string, tlsConfig *tlsutil.TLSConfig) Factory {
+
+	if tlsConfig != nil && !tlsConfig.HasCertAuth() {
```

Comment:
what does else mean?

Reply:
`tlsConfig == nil` means that TLS encryption is disabled.
`tlsConfig.HasCertAuth()` means that CA certificate is manually specified for server authentication.

So, the `else` clause here means either of the above cases.

Reply:
Thanks Yohei, I just realized `manually specified` is existing code.